### PR TITLE
Add feature to allow override on timestamp and signature

### PR DESCRIPTION
**Pull request by leptians**
To allow override on timestamp and signature on API call level

### CODE REVIEW

**Comment by asisayag2** [comment]
@strausr , let's review and make sure it goes into the next version

File: lib-es5/utils/index.js [hunk 1]
```diff
@@ -1063,6 +1063,8 @@ function process_request_params(params, options) {
   if (options.unsigned != null && options.unsigned) {
     params = exports.clear_blank(params);
     delete params.timestamp;
+  } else if (options.signature ) {
+    params = exports.clear_blank(options);
```
Comment (strausr) [inline lib-es5/utils/index.js]: Can you please explain why you added the clear_blank?

File: lib-es5/utils/index.js [hunk 2]
```diff
@@ -354,7 +354,7 @@ function build_upload_params(options) {
     quality_analysis: utils.as_safe_bool(options.quality_analysis),
     responsive_breakpoints: utils.generate_responsive_breakpoints_string(options.responsive_breakpoints),
     return_delete_token: utils.as_safe_bool(options.return_delete_token),
-    timestamp: exports.timestamp(),
+    timestamp: options.timestamp ? options.timestamp : exports.timestamp(),
```
Comment (strausr) [inline lib-es5/utils/index.js]: Can be changed into options.timestamp || exports.timestamp()

File: lib-es5/utils/index.js [hunk 3]
```diff
@@ -1073,7 +1075,7 @@ function private_download_url(public_id, format) {
   var options = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : {};
 
   var params = exports.sign_request({
-    timestamp: exports.timestamp(),
+    timestamp: options.timestamp ? options.timestamp : exports.timestamp(),
```
Comment (strausr) [inline lib-es5/utils/index.js]: Can be changed into options.timestamp || exports.timestamp()

File: lib-es5/utils/index.js [hunk 4]
```diff
@@ -1092,7 +1094,7 @@ function zip_download_url(tag) {
   var options = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
 
   var params = exports.sign_request({
-    timestamp: exports.timestamp(),
+    timestamp: options.timestamp ? options.timestamp : exports.timestamp(),
```
Comment (strausr) [inline lib-es5/utils/index.js]: Can be changed into options.timestamp || exports.timestamp()

File: test/utils_spec.js [hunk 5]
```diff
@@ -1411,6 +1425,47 @@ describe("utils", function () {
         api_key: "1234",
       });
     });
+    it("should allow timestamp in sign_request", function() {
```
Comment (strausr) [inline test/utils_spec.js]: The tests should test exposed functions. Can you please test Upload and verify that it is passing the signature and timestamp?

File: lib-es5/utils/index.js (hunk 1 above)
Comment (strausr) [inline lib-es5/utils/index.js]: Can you please explain what clear_blank is being used?

Comment (leptians) [reply lib-es5/utils/index.js]: Following the same practice the same as from line#1064

File: test/utils_spec.js (hunk 5 above)
Comment (leptians) [reply test/utils_spec.js]: @brian-cloudinary, can you please complete this task?

**Pull request merged**